morse_decode.rb

```diff
@@ -0,0 +1,60 @@
+MORSE_DICT = {
+  '.-' => 'a',
+  '-...' => 'b',
+  '-.-.' => 'c',
+  '-..' => 'd',
+  '.' => 'e',
+  '..-.' => 'f',
+  '--.' => 'g',
+  '....' => 'h',
+  '..' => 'i',
+  '.---' => 'j',
+  '-.-' => 'k',
+  '.-..' => 'l',
+  '--' => 'm',
+  '-.' => 'n',
+  '---' => 'o',
+  '.--.' => 'p',
+  '--.-' => 'q',
+  '.-.' => 'r',
+  '...' => 's',
+  '-' => 't',
+  '..-' => 'u',
+  '...-' => 'v',
+  '.--' => 'w',
+  '-..-' => 'x',
+  '-.--' => 'y',
+  '--..' => 'z',
+  ' ' => ' ',
+  '.----' => '1',
+  '..---' => '2',
+  '...--' => '3',
+  '....-' => '4',
+  '.....' => '5',
+  '-....' => '6',
+  '--...' => '7',
+  '---..' => '8',
+  '----.' => '9',
+  '-----' => '0'
+}.freeze
+
+def decode_char(char, dict: MORSE_DICT)
+  dict[char].upcase
+end
+
+def decode_word(word)
+  result = ''
+
+  word.split(/ /).each do |letter|
+    result += decode_char(letter)
+  end
+  result
+end
+
+def decode(sentence)
+  result = ''
+  sentence.split(/   /).each do |word|
+    result += " #{decode_word(word)}"
+  end
+  result.strip
+end
```

Comment:
- [ ] Good job so far. However, I got an error when I tried to print the results from your methods. See the error in the image below. Kindly fix it.
<img width="565" alt="chrome_vw85qytlri" src="https://user-images.githubusercontent.com/81468675/166908227-6f2372bb-b786-4106-aad3-aa05701f250b.png">

Reply:
Thank you @julie-ify. How do you get this error? We don't get it.

![image](https://user-images.githubusercontent.com/11241315/166909955-8d90c926-7986-4884-9dc1-9810572670f3.png)

Reply:
Hi @julie-ify 
Can please share with us the command that you used to run the file.
Cheers

Reply:
Here I made a demo of how it works for me:
![demo](https://user-images.githubusercontent.com/51741471/166913580-277e4dba-54e4-4f3e-a4a1-b3d2a16f86c3.gif)

Reply:
Found the issue: unauthorized input.

Reply:
hi @Sboursen and @ahangarha. I added this 👇 at the end of the file and then `ruby morse_decode.rb` to run the file.
```ruby
p decode_char('.-')
p decode_word('-- -.--')
p decode("-- -.--   -. .- -- .")
p decode(".-   -... --- -..-   ..-. ..- .-.. .-..    --- ..-.    .-. ..- -... .. . ...")

```

Reply:
One of the morse code are wrong. That is why we get error. Because we try to run `upcase` on some thing that doesn't exist.

Maybe it was unintended but we found a bug and fixed it.

Thanks